src/utils.rs

```diff
@@ -127,27 +126,27 @@ mod tests {
         let second_closest = close_nodes[1];
         let mut combined: Vec<u8> = Vec::new();
 
-        for i in original_name.0.into_iter() {
+        for i in &original_name.0 {
```

Comment:
It's the same thing we were just discussing - IntoIterator being only implemented for slices and not for arrays.